icicle/appUtils/msm/msm.cu

```diff
@@ -636,66 +649,73 @@ namespace msm {
 #endif
 
       P* d_final_result;
-      if (!is_result_on_device) CHK_IF_RETURN(cudaMallocAsync(&d_final_result, sizeof(P), stream));
+      if (!are_results_on_device) CHK_IF_RETURN(cudaMallocAsync(&d_final_result, sizeof(P) * batch_size, stream));
 
       P* final_results;
-      if (is_big_triangle) {
-        CHK_IF_RETURN(cudaMallocAsync(&final_results, sizeof(P) * nof_bms, stream));
+      if (is_big_triangle || c == 1) {
+        CHK_IF_RETURN(cudaMallocAsync(&final_results, sizeof(P) * nof_bms * batch_size, stream));
```

Comment:
this seems to be used a lot, maybe define total_nof_bms = nof_bms*batch_size

Reply:
ok